ncat/ncat_connect.c

```diff
@@ -788,6 +788,8 @@ static int do_proxy_socks5(void)
             goto error;
     }
 
+    free(proxy_auth);
```

Comment:
This will cause a double free() if you jump to ```error:``` from now on.  Either put this bellow all the gotos, or put ```proxy_auth = NULL;``` immediately after the first free().